src/plus/integrations/authentication/jira.ts

```diff
@@ -1,111 +1,70 @@
+import { IssueIntegrationId } from '../providers/models';
+import type { CloudIntegrationsApi } from './cloudIntegrationsApi';
 import type {
 	IntegrationAuthenticationProvider,
 	IntegrationAuthenticationSessionDescriptor,
 } from './integrationAuthentication';
 import type { ProviderAuthenticationSession } from './models';
 
-type ConnectionType = 'oauth' | 'personal_access_token';
-
-type ProviderTokenData = {
-	accessToken: string;
-	type: ConnectionType;
-	domain: string;
-	expiresIn: number;
-	scopes: string;
-};
-
-type ProviderAuthorizationData = {
-	url: string;
-};
-
-/* type ProviderRsp<T> = {
-	data: T | null;
-	error: string | null;
-}; */
-
-type providerInput = 'jira' | 'trello' | 'gitlab' | 'github' | 'bitbucket' | 'azure';
-
 export class JiraAuthenticationProvider implements IntegrationAuthenticationProvider {
-	constructor(private readonly connection: ServerConnection) {}
+	constructor(
+		private readonly container: Container,
+		private readonly getCloudIntegrationsApi: () => Promise<CloudIntegrationsApi>,
+	) {}
 
-	private readonly authProviderId = 'jira';
+	private readonly authProviderId = IssueIntegrationId.Jira;
 
 	getSessionId(descriptor?: IntegrationAuthenticationSessionDescriptor): string {
 		return descriptor?.domain ?? '';
 	}
 
-	private async getTokenData(
-		provider: providerInput,
-		refresh: boolean = false,
-	): Promise<ProviderTokenData | undefined> {
-		const tokenRsp = await this.connection.fetchGkDevApi(
-			`v1/provider-tokens/${provider}${refresh ? '/refresh' : ''}`,
-			{ method: refresh ? 'POST' : 'GET' },
-		);
-		if (!tokenRsp.ok) {
-			// TODO: Handle errors
-			const error = (await tokenRsp.json())?.error;
-			if (error != null) {
-				Logger.error(`Failed to ${refresh ? 'refresh' : 'get'} ${provider} token from cloud: ${error}`);
-			}
-			return undefined;
-		}
-
-		return (await tokenRsp.json())?.data as Promise<ProviderTokenData | undefined>;
-	}
-
-	private async authorize(provider: providerInput): Promise<ProviderAuthorizationData | undefined> {
-		const authorizeRsp = await this.connection.fetchGkDevApi(
-			`v1/provider-tokens/${provider}/authorize`,
-			{
-				method: 'GET',
-			},
-			{
-				query: 'source=gitlens',
-			},
-		);
-		if (!authorizeRsp.ok) {
-			// TODO: HANDLE ERROR
-			const error = (await authorizeRsp.json())?.error;
-			if (error != null) {
-				Logger.error(`Failed to authorize with ${provider}: ${error}`);
-			}
-			return undefined;
-		}
-
-		return (await authorizeRsp.json())?.data as Promise<ProviderAuthorizationData | undefined>;
-	}
-
 	async createSession(
 		descriptor?: IntegrationAuthenticationSessionDescriptor,
+		options?: { authorizeIfNeeded?: boolean },
 	): Promise<ProviderAuthenticationSession | undefined> {
-		let tokenData = await this.getTokenData(this.authProviderId);
+		const api = await this.getCloudIntegrationsApi();
+		let tokenData = await api.getTokenData(this.authProviderId);
```

Comment:
Should we move the "refreshing" into this call (on a flag) or should we keep the control at the provider level?

Reply:
The "refresh" is currently automatically performed in this call when your retrieved token is expired. I don't see a need to ever have that not be the case, so I'm not sure if we even need it manually controlled.